sdk-ingestion/src/main/java/care/data4life/sdk/Data4LifeClient.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) 2020 D4L data4life gGmbH / All rights reserved.
+ *
+ * D4L owns all legal rights, title and interest in and to the Software Development Kit ("SDK"),
+ * including any intellectual property rights that subsist in the SDK.
+ *
+ * The SDK and its documentation may be accessed and used for viewing/review purposes only.
+ * Any usage of the SDK for other purposes, including usage for the development of
+ * applications/third-party applications shall require the conclusion of a license agreement
+ * between you and D4L.
+ *
+ * If you are interested in licensing the SDK for your own applications/third-party
+ * applications and/or if you’d like to contribute to the development of the SDK, please
+ * contact D4L by email to help@data4life.care.
+ */
+
+package care.data4life.sdk;
```

Comment:
Why is it in this package?
This create a clash with the other SDKs.

For instance the test-broker is using the Data4LifeClient from the jvm-core and the ingestion-sdk at the same time.

Reply:
@salehgamil

Reply:
I feel, that having the same class (`care.data4life.sdk.Data4LifeClient`) with different signatures in different jars is not very good practice, is it?

How about moving this into `care.data4life.ingestion-sdk` ??

Reply:
I believe keeping the name was actually a request from Wolf. Currently, the sdk-android and sdk-jvm packages also both define classes called `Data4LifeClient` in the `care.data4life.sdk` package, and they also have different signatures from the constructors & the `init()` factory methods. So it seems the idea is to have the package be the same and just binding in the right JAR. But for my sake, we could arrange it differently.

Reply:
@aps do you have some background on this?
For us it would be really benefitial to have the ingestion SDK's Data4LifeClient in a different package.

Reply:
Yes as @me-d4l is saying, since the project is meant to be a multiplatform project the android and jvm classes will have the same package. 

I don't know what the purpose of the `ingestion-sdk` is, but if it's seen as a new platform then the package should stay the same for the consistency reasons. 

> For instance the test-broker is using the Data4LifeClient from the jvm-core and the ingestion-sdk at the same time.

Why do you want to have both of them in there?

Reply:
It's only for testing purposes. 
The ingestion SDK has only limited capabilities (append only) and with the JVM SDK we are testing if the uploaded resources are there.
Right now, in the very same spring application are the two SDKs reachable on different endpoints.
Of course we could extract that and put this into different spring applications but it is a but cumbersome.

But well, if this is the way you want to have it, than @salehgamil can look into the changes for the test-broker next week. 😬

Reply:
I just double checked, but it actually cannot get moved to another package. All the core components are in the `.sdk` package which won't be accessible from the `.integration-sdk` package due to the visibility being package-private.

Reply:
Ok, thanks for looking into this. Then we must change it on our site.
🙏🏻

Reply:
The library is not designed to be used in different flavors alongside in the same project.  But an  idea: you could have a `Data4LifeIngestionClient` that implements `BaseClient` to avoid collisions in the `.sdk`package.